StudioCore/MsbEditor/Entity.cs

```diff
@@ -451,6 +451,20 @@ public void BuildReferenceMap()
                     {
                         if (p.PropertyType.IsArray)
                         {
+                            Array array = (Array)p.GetValue(WrappedObject);
+                            foreach (var i in array)
+                            {
+                                var sref = (string)i;
+                                if (sref != null && sref != "")
+                                {
+                                    var obj = Container.GetObjectByName(sref);
```

Comment:
This is effectively O(n^2), so a more scalable linear solution might be better (guess we might need to maintain a hashmap of names or something).

Reply:
I don't know enough to make a proper solution. This does serve as a functional solution (I was capable of doing) that lets the system work in a larger majority of situations, and it has behaved fast & fine in every common case I tested. 
Ref scanning/MSB search aren't able to go through structs as-is, so bigger changes need to happen in the future anyway and a rewrite at that time might be more appropriate.